### PR TITLE
Plans: Add new copy for tier 1 & 2 storage upgrades

### PR DESCRIPTION
**Pull request by mattgawarecki**
Resolves `1200412004370260-as-1201090224957164`.

#### Changes proposed in this Pull Request

* Create new product card copy (headers, feature items, etc.) for storage upgrade "products."
* Create new hooks to determine how (and which) storage upgrades should be displayed for a given site.
* Create a selector to retrieve all of a given site's purchases that are eligible for upgradable storage.
* Modify the storage upgrade pricing page to use the new cards and copy.
* Refactor `getSlugInTerm` so that it can address more use cases.
* Remove the `getTieredProducts` hook, as it is no longer in use.

##### Known issues

* Feature highlighting is not enabled yet, but will be addressed in #56595.
* Some page styles are incorrect on narrow screens. We'll address this in a separate PR.

#### Testing instructions

**NOTE:** The `jetpack/only-realtime-products` feature flag must be enabled during the entire testing process. Append `?flags=jetpack/only-realtime-products` to your browser URL, or enable the flag in your environment's Calypso configuration file.

In both Calypso Blue and Calypso Green, test the following:

1. Select a site with no purchases that qualify for storage upgrades, then visit `/pricing/storage/{siteSlug}`. You should see a screen that resembles the below screenshot. Verify all copy looks correct, all links redirect to checkout for tiered Backup products, and all links are correct for the billing term you have selected (i.e., Monthly or Yearly).
2. Select a site with a NON-TIERED Backup or Security purchase (e.g., Backup Daily, Security Real-time). When you visit `/pricing/storage/{siteSlug}`, you should see the same page as in step 1, and it should function identically.

Screen Shot 2021-10-01 at 10 57 23

3. Select a site with a TIER 1 Backup (Monthly) purchase. When you visit `/pricing/storage/{siteSlug}`, you should see a screen that resembles the below screenshot. Verify:
  a. All copy looks correct;
  b. The 20GB card is displayed with "You own this product;"
  c. Toggling the billing term to Monthly changes the item pricing, and changes 20GB call-to-action link to say "Manage Subscription" and sends you to the Manage Purchase page for your purchase;
  d. Toggling the billing term to Yearly changes the item pricing, and changes the 20GB call-to-action button to say "Upgrade to Yearly" and sends you to checkout for Backup 20GB with a yearly billing term;
  e. The 1TB card is displayed with the price for Backup 1TB;
  f. Clicking the 1TB card's call-to-action link sends you to checkout for Backup (1TB), with the billing term you selected.

Screen Shot 2021-10-01 at 11 01 03

4. Select a site with a TIER 1 Security (Monthly) purchase. When you visit `/pricing/storage/{siteSlug}`, you should see a screen that resembles the screen from step 3. Verify:
  a. Toggling the billing term to Monthly changes the item pricing, and changes the 20GB card's call-to-action link to say "Manage Subscription" and sends you to the Manage Purchase page for your Security purchase;
  b. Toggling the billing term to Yearly changes the item pricing, and changes the 20GB card's call-to-action link to say "Upgrade to Yearly" and sends you to checkout for Security 20GB with a yearly billing term;
  c. The 1TB card is displayed with the price for Security (1TB), and its call-to-action link sends you to checkout for Security (1TB), with the billing term you selected.

5. Select a site with a TIER 2 Monthly purchase (either Backup or Security). When you visit `/pricing/storage/{siteSlug}`, you should see a screen that resembles the below screenshot. Verify:
  a. The 20GB card is not displayed;
  b. The 1TB card is displayed with "You own this product";
  c. Toggling the billing term to Monthly changes the 1TB card's call-to-action link to say "Manage Subscription" and sends you to the Manage Purchase page for the product you purchased;
  d. Toggling the billing term to Yearly changes the 1TB card's call-to-action link to say "Upgrade to Yearly" and sends you to checkout for the yearly version of the 1TB product you originally purchased;

image

### CODE REVIEW

File: client/lib/plans/features-list.js [hunk 1]
```diff
@@ -1073,7 +1087,7 @@ export const FEATURES_LIST = {
 
 	[ FEATURE_PRODUCT_SEARCH_V2 ]: {
 		getSlug: () => FEATURE_PRODUCT_SEARCH_V2,
-		getTitle: () => i18n.translate( 'Site Search: up to 100k records' ),
+		getTitle: () => i18n.translate( 'Search up to 100k records' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/lib/plans/features-list.js [hunk 2]
```diff
@@ -1102,7 +1116,7 @@ export const FEATURES_LIST = {
 
 	[ FEATURE_CRM_V2 ]: {
 		getSlug: () => FEATURE_CRM_V2,
-		getTitle: () => i18n.translate( 'CRM: Entrepreneur bundle' ),
+		getTitle: () => i18n.translate( 'CRM Entrepreneur bundle' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **8** times:
`translate( 'CRM: Entrepreneur' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: client/lib/plans/features-list.js [hunk 3]
```diff
@@ -1314,6 +1328,67 @@ export const FEATURES_LIST = {
...
+		getSlug: () => FEATURE_JETPACK_ONE_CLICK_RESTORES_1_YEAR,
+		getTitle: () => i18n.translate( 'One-click restore from the last year' ),
+	},
+	[ FEATURE_JETPACK_ONE_CLICK_RESTORES_UNLIMITED ]: {
+		getSlug: () => FEATURE_JETPACK_ONE_CLICK_RESTORES_UNLIMITED,
+		getTitle: () => i18n.translate( 'One-click restores from any time' ),
+	},
+	[ FEATURE_JETPACK_PRODUCT_BACKUP ]: {
+		getSlug: () => FEATURE_JETPACK_PRODUCT_BACKUP,
+		getTitle: () => i18n.translate( 'All Backup features' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'All Jetpack Backup features' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: client/lib/plans/features-list.js [hunk 4]
```diff
@@ -1314,6 +1328,67 @@ export const FEATURES_LIST = {
...
+		isHighlighted: () => true,
+	},
+	[ FEATURE_JETPACK_PRODUCT_SCAN ]: {
+		getSlug: () => FEATURE_JETPACK_PRODUCT_SCAN,
+		getTitle: () => i18n.translate( 'Scan (real-time & automated)' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Scan (real-time, automated)' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: client/lib/plans/features-list.js [hunk 5]
```diff
@@ -1314,6 +1328,67 @@ export const FEATURES_LIST = {
...
+		isHighlighted: () => true,
+	},
+	[ FEATURE_JETPACK_PRODUCT_SECURITY ]: {
+		getSlug: () => FEATURE_JETPACK_PRODUCT_SECURITY,
+		getTitle: () => i18n.translate( 'All Security features' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'All Security Daily features' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+						{
+							text: translate( '20GB backup storage' ),
+							isHighlighted: true,
+						},
+						{
+							text: translate( 'One-click restore from the last 30 days of backups' ),
+							isHighlighted: true,
+						},
+						{
+							text: translate( '30-day activity log' ),
```
Comment (a8ci18n) [inline client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts:37]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Activity log: 30-day archive' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+							isHighlighted: true,
+						},
+						{
+							text: translate( 'Real-time backups (as you edit)' ),
```
Comment (a8ci18n) [inline client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts:41]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Real-time backups &mdash; we save as you edit' )` *ES Score: 6*

File: client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+				displayName: translate( '1TB' ),
+				subheader: translate( 'of backup storage' ),
+				buttonLabel: translate( 'Upgrade storage' ),
+				description: translate(
+					'Go back in time and recover all your information for up to a year, with 1TB storage space.'
+				),
+				features: {
+					items: [
+						{
+							text: translate( '1TB backup storage' ),
```
Comment (a8ci18n) [inline client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Backup (1TB)', { context: 'Jetpack Backup Solution'} )` *ES Score: 6*

**Comment by rcanepa** [comment]
There is one unit test failing:
image

File: client/lib/plans/features-list.js [hunk 9]
```diff
@@ -1314,6 +1328,67 @@ export const FEATURES_LIST = {
...
+		getSlug: () => FEATURE_JETPACK_CLOUD_BASED_BACKUPS,
+		getTitle: () => i18n.translate( 'Backups are cloud based' ),
+	},
+	[ FEATURE_JETPACK_20GB_BACKUP_STORAGE ]: {
+		getSlug: () => FEATURE_JETPACK_20GB_BACKUP_STORAGE,
+		getTitle: () => i18n.translate( '20GB of backup storage' ),
+	},
+	[ FEATURE_JETPACK_1TB_BACKUP_STORAGE ]: {
+		getSlug: () => FEATURE_JETPACK_1TB_BACKUP_STORAGE,
+		getTitle: () => i18n.translate( '1TB of backup storage' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **2** times:
`translate( '1TB of storage' )` *ES Score: 7*

**Comment by mattgawarecki** [comment]
> There is one unit test failing: image

Argh, you're right! I forgot about those functions. Fixing the broken bits now. 👍

**Comment by robertf4** [comment]
This needs a rebase due to the merge of https://github.com/Automattic/wp-calypso/pull/56552.

**Comment by mattgawarecki** [comment]
> This needs a rebase due to the merge of #56552.

Absolutely! Done.

**Review by ederrengifo — CHANGES_REQUESTED** [review]
A couple of issues were found:

- When the customer reaches the TIER 2 purchase (Step 5) and they only see one column, that layout breaks the footer:
Screen Shot 2021-10-05 at 10 28 37
- I don't see the "Save 20%" label next to the "Bill yearly" option from the designs. We might have decided to remove it at some point so would be good to investigate if that was the case.

**Comment by mattgawarecki** [comment]
> A couple of issues were found:

@ederrengifo Thanks for letting me know about these, and for making sure they're documented. I'm happy to report these are actually addressed in two separate PRs, since it turns out they're not immediately due to any changes in this one:

>     * When the customer reaches the TIER 2 purchase (Step 5) and they only see one column, that layout breaks the footer:

This will be resolved in #56802.

> - I don't see the "Save 20%" label next to the "Bill yearly" option from the designs. We might have decided to remove it at some point so would be good to investigate if that was the case.

We shipped a fix for this just moments ago, in #56801. 👍

File: client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+			( {
+				...slugToSelectorProduct( slug ),
+				displayName: translate( '10GB', { comment: 'As in, "10GB of backup storage"' } ),
+				subheader: translate( 'of backup storage' ),
+				buttonLabel: translate( 'Upgrade storage' ),
+				// description: <the default description of the product being upgraded>,
+				features: {
+					items: [
+						{
+							text: translate( '10GB backup storage' ),
```
Comment (a8ci18n) [inline client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( '10GB storage' )` *ES Score: 6*

**Review by robertf4 — APPROVED** [review]
Looks good to me, and tests well.

File: client/lib/plans/features-list.js [hunk 11]
```diff
@@ -1078,7 +1089,7 @@ export const FEATURES_LIST = {
 
 	[ FEATURE_PRODUCT_SEARCH_V2 ]: {
 		getSlug: () => FEATURE_PRODUCT_SEARCH_V2,
-		getTitle: () => i18n.translate( 'Site Search: up to 100k records' ),
+		getTitle: () => i18n.translate( 'Site Search up to 100k records' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Search: up to 100k records' )` *ES Score: 10*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: client/lib/plans/features-list.js [hunk 12]
```diff
@@ -1344,6 +1355,47 @@ export const FEATURES_LIST = {
...
+		getSlug: () => FEATURE_JETPACK_10GB_BACKUP_STORAGE,
+		getTitle: () => i18n.translate( '10GB of backup storage' ),
+	},
+	[ FEATURE_JETPACK_1TB_BACKUP_STORAGE ]: {
+		getSlug: () => FEATURE_JETPACK_1TB_BACKUP_STORAGE,
+		getTitle: () => i18n.translate( '1TB of backup storage' ),
+	},
+	[ FEATURE_JETPACK_30_DAY_ARCHIVE_ACTIVITY_LOG ]: {
+		getSlug: () => FEATURE_JETPACK_30_DAY_ARCHIVE_ACTIVITY_LOG,
+		getTitle: () => i18n.translate( '30-day archive & activity log' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Activity log: 30-day archive' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: client/lib/plans/features-list.js [hunk 13]
```diff
@@ -1344,6 +1355,47 @@ export const FEATURES_LIST = {
...
+	},
+	[ FEATURE_JETPACK_1_YEAR_ARCHIVE_ACTIVITY_LOG ]: {
+		getSlug: () => FEATURE_JETPACK_1_YEAR_ARCHIVE_ACTIVITY_LOG,
+		getTitle: () => i18n.translate( '1-year archive & activity log' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Activity log: 1-year archive' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: client/lib/plans/features-list.js [hunk 14]
```diff
@@ -1344,6 +1355,47 @@ export const FEATURES_LIST = {
...
+		getSlug: () => FEATURE_JETPACK_PRODUCT_VIDEOPRESS,
+		getTitle: () => i18n.translate( 'VideoPress' ),
+	},
+	[ FEATURE_JETPACK_ALL_BACKUP_SECURITY_FEATURES ]: {
+		getSlug: () => FEATURE_JETPACK_ALL_BACKUP_SECURITY_FEATURES,
+		getTitle: () => i18n.translate( 'All Backup & Security features' ),
+	},
+	[ FEATURE_JETPACK_REAL_TIME_CLOUD_BACKUPS ]: {
+		getSlug: () => FEATURE_JETPACK_REAL_TIME_CLOUD_BACKUPS,
+		getTitle: () => i18n.translate( 'Real-time cloud backups' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Real-time backups' )` *ES Score: 6*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: client/lib/plans/features-list.js [hunk 15]
```diff
@@ -1344,6 +1355,47 @@ export const FEATURES_LIST = {
...
+	},
+	[ FEATURE_JETPACK_REAL_TIME_MALWARE_SCANNING ]: {
+		getSlug: () => FEATURE_JETPACK_REAL_TIME_MALWARE_SCANNING,
+		getTitle: () => i18n.translate( 'Real-time malware scanning' ),
```
Comment (a8ci18n) [inline client/lib/plans/features-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Real-Time Malware Scanning' )` *ES Score: 13*

File: packages/calypso-products/src/plans-list.js [hunk 16]
```diff
@@ -845,14 +849,12 @@ const getPlanJetpackSecurityT1Details = () => ( {
 	availableFor: ( plan ) => [ PLAN_JETPACK_FREE, ...JETPACK_LEGACY_PLANS ].includes( plan ),
 	getDescription: () =>
 		translate(
-			'Get next-level protection with real-time backups, real-time scan and all essential security tools.'
+			'Easy-to-use, comprehensive WordPress site security including backups, malware scanning, and spam protection.'
```
Comment (a8ci18n) [inline packages/calypso-products/src/plans-list.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Jetpack Security provides easy-to-use, comprehensive WordPress site security including backups, malware scanning, and spam protection' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

File: packages/calypso-products/src/translations.js [hunk 17]
```diff
@@ -244,7 +244,9 @@ export const getJetpackProductsDescriptions = () => {
 	const backupRealtimeDescription = translate(
 		'Real-time backups save every change and one-click restores get you back online quickly.'
 	);
-	const backupT1Description = backupDailyDescription;
+	const backupT1Description = translate(
+		'Save every change with real-time backups and get back online quickly with one-click restores.'
```
Comment (a8ci18n) [inline packages/calypso-products/src/translations.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Real-time backups save every change and one-click restores get you back online quickly' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/storage-upgrade-cards-copy)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6761129

Thank you @mattgawarecki for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6761129).

**Pull request merged**